ext/uri/php_uri.c

```diff
@@ -60,18 +615,34 @@ static PHP_MINFO_FUNCTION(uri)
 
 static PHP_MSHUTDOWN_FUNCTION(uri)
 {
+	zend_hash_destroy(&uri_handlers);
 
 	return SUCCESS;
 }
 
 PHP_RINIT_FUNCTION(uri)
 {
+	uri_handler_t *handler;
+
+	ZEND_HASH_MAP_FOREACH_PTR(&uri_handlers, handler) {
+		if (handler->init_parser() == FAILURE) {
+			return FAILURE;
+		}
+	} ZEND_HASH_FOREACH_END();
```

Comment:
Looking at the code, calling `init_parser()` multiple times for the Lexbor implementation is unsafe. To be this is an indicator that we should not have the `init_parser()` function as part of the `uri_handler_t`: It is not meant to be called by anyone else!

Instead I would suggest `RINIT` calls the initialization logic directly. If an extension wants to add their own URI handler, they can add their own `RINIT` callback:

```suggestion
	if (lexbor_init_parser() == FAILURE) {
		return FAILURE;	
	}
```

Reply:
Yes, and I realized that some of the init code could rather be in the MINIT. And similarly, neither the `destroy_parser` handler is actually needed, so I also got rid of it.